src/openforms/middleware.py

```diff
@@ -49,3 +50,28 @@ def __call__(self, request):
 
         response[CSRF_TOKEN_HEADER_NAME] = get_token(request)
         return response
+
+
+class CanNavigateBetweenStepsMiddleware:
+    """
+    Add a header to the response to let the frontend know if the user is allowed to navigate between submission steps
+    that have not been completed yet.
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
+
+        # Only add the header if it's an api endpoint
+        if not request.path.startswith("/api"):
+            return response
+
+        response[CAN_NAVIGATE_BETWEEN_STEPS_HEADER_NAME] = "false"
+
+        user = request.user
+        if user.is_staff and user.has_perm("forms.change_form"):
+            response[CAN_NAVIGATE_BETWEEN_STEPS_HEADER_NAME] = "true"
```

Comment:
I forgot about the permission check - maybe we should rename it to `X-Is-Form-Designer` rather than `X-Is-Staff` after all :thinking:

Reply:
Yes I think `X-Is-Form-Designer` fits better